### PR TITLE
Refactor CConsoleCommands::Debugscript

### PR DESCRIPTION
**Pull request by Pirulax**
In a previous commit ( 8f25c22 ) the `debugscript` command got fixed, but I think it needs a refactor as well, so here we are.
I didnt want to be consistent with the existing style, because I think this is more readable, but correct me if im wrong.

### CODE REVIEW

**Comment by Pirulax** [comment]
> yeah I see, Im meant the || strlen..

Ah well, thats necessary to make sure I actually have a valid char when doing that `atoi` like thing.

File: Client/game_sa/CAEVehicleAudioEntitySA.cpp [hunk 1]
```diff
@@ -53,15 +55,80 @@ void CAEVehicleAudioEntitySA::TurnOnRadioForVehicle()
...
+
+void CAEVehicleAudioEntitySA::CancelVehicleEngineSound(short engineSoundStateID)
+{
+    return ((void(__thiscall*)(void*, short))FUNC_CAEVehicleAudioEntity__CancelVehicleEngineSound)((void*)m_pInterface, engineSoundStateID);
+}
+
+void CAEVehicleAudioEntitySA::MakeSureEngineSoundsAreLoaded()
+{
+    // TODO: Check if this is the right way to do this
+    //       (aka should we write a function to load the sound bank, and all that, or this hack is just fine)
```
Comment (ghost) [inline Client/game_sa/CAEVehicleAudioEntitySA.cpp:86]:
```suggestion
    // CAEVehicleAudioEntity::JustGotInVehicleAsDriver loads the engine sound banks
```
If it works and has no side effects, then it's totally fine.

File: Client/game_sa/CAEVehicleAudioEntitySA.cpp [hunk 2]
```diff
@@ -53,15 +55,80 @@ void CAEVehicleAudioEntitySA::TurnOnRadioForVehicle()
...
+    ((void(__thiscall*)(void*))FUNC_CAEVehicleAudioEntity__JustGotInVehicleAsDriver)((void*)m_pInterface);
+}
+
+unsigned int CAEVehicleAudioEntitySA::SetEngineAccelerationSoundBank(unsigned int bankId) // Seems to crash with id 83
+{
+    if (bankId == m_pInterface->m_wEngineOnSoundBankId)
+        return m_pInterface->m_wEngineOnSoundBankId;
+
+    // TODO: Add check for bankid (maybe use CAEAudioHardware or something)
+    auto oldValue = m_pInterface->m_wEngineOnSoundBankId;
```
Comment (ghost) [inline Client/game_sa/CAEVehicleAudioEntitySA.cpp:96]:
```suggestion
    short oldValue = m_pInterface->m_wEngineOnSoundBankId;
```
The auto keyword is mostly used to improve readability. This is not good use of auto. I personally only use it when the line takes way too much space and makes the code hard to read.

File: Client/game_sa/CAEVehicleAudioEntitySA.h [hunk 3]
```diff
@@ -122,27 +122,74 @@
 #define FUNC_CAEVehicleAudioEntity__Service                                            0x502280
 
 #define VAR_CAEVehicleAudioEntity__s_pPlayerDriver                                     0xB6B990
+#define VAR_CAEVehicleAudioEntity__s_aDefaultVehicleAudioSettings                      0x860AF0
+#define VAR_CAEVehicleAudioEntity__s_aDummyEngineSlots                                 0xB6B9A0
```
Comment (ghost) [inline Client/game_sa/CAEVehicleAudioEntitySA.h:126]: There's no reason to define preprocessors when you can just easily write
```
// header file
const int MAX_DUMMY_ENGINE_SLOTS = 10;
class CAEVehicleAudioEntitySA 
{
public:
    static DummyEngineSlot(&s_DummyEngineSlots)[MAX_DUMMY_ENGINE_SLOTS];
};
```
```
// .cpp file
DummyEngineSlot(&CAEVehicleAudioEntitySA::s_DummyEngineSlots)[MAX_DUMMY_ENGINE_SLOTS] = *(DummyEngineSlot(*)[MAX_DUMMY_ENGINE_SLOTS])0xB6B9A0;
```

**Review by ghost — COMMENTED** [review]
I've pointed out a few things that must be addressed before we can merge. Overall, the PR looks good. We just need to test it.

**Review by qaisjp — CHANGES_REQUESTED** [review]
This PR contains a lot more than the title? I am confusion. I think 3bfb3a87b34fb420b22340340c547dcfc1a6a190 needs to be removed?

**Comment by Pirulax** [comment]
Yeah, i forgot that I have opened a PR for this branch, so I casually pushed my audio entity things here(because I forgot to switch branches)..
Ill revert it. (Well, soon (TM))

File: Client/game_sa/CAEVehicleAudioEntitySA.h (hunk 3 above)
Comment (Pirulax) [reply Client/game_sa/CAEVehicleAudioEntitySA.h:126]: I know about `constexpr` (and I use it a lot), but I just wanted to be consistent about using macros.
I might as well just copy paste the address directly then.

**Comment by Pirulax** [comment]
Issue: Contains code changes unrelated to PR.

**Pull request closed without merging**